internal/cmd/ping.go

```diff
@@ -0,0 +1,150 @@
+package cmd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/pion/webrtc/v3"
+	"github.com/spf13/cobra"
+	"nhooyr.io/websocket"
+
+	"cdr.dev/coder-cli/coder-sdk"
+	"cdr.dev/coder-cli/pkg/clog"
+	"cdr.dev/coder-cli/wsnet"
+)
+
+func pingCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "ping [workspace_name]",
+		Short:   "Ping a Coder workspace",
+		Example: `coder ping my-dev`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+			client, err := newClient(ctx, true)
+			if err != nil {
+				return err
+			}
+			workspace, err := findWorkspace(ctx, client, args[0], coder.Me)
+			if err != nil {
+				return err
+			}
+			if workspace.LatestStat.ContainerStatus != coder.WorkspaceOn {
+				return clog.Error("workspace not available",
+					fmt.Sprintf("current status: \"%s\"", workspace.LatestStat.ContainerStatus),
+					clog.BlankLine,
+					clog.Tipf("use \"coder workspaces rebuild %s\" to rebuild this workspace", workspace.Name),
+				)
+			}
+			servers, err := client.ICEServers(ctx)
+			if err != nil {
+				return err
+			}
+			url := client.BaseURL()
+			connectionStart := time.Now()
+			dialer, err := wsnet.DialWebsocket(ctx, wsnet.ConnectEndpoint(&url, workspace.ID, client.Token()), &wsnet.DialOptions{
+				ICEServers:         servers,
+				TURNProxyAuthToken: client.Token(),
+				TURNRemoteProxyURL: &url,
+				TURNLocalProxyURL:  &url,
+			}, &websocket.DialOptions{})
+			if err != nil {
+				return err
+			}
+			connectionMS := float64(time.Since(connectionStart).Microseconds()) / 1000
+			candidates, err := dialer.Candidates()
+			if err != nil {
+				return err
+			}
+			relay := candidates.Local.Typ == webrtc.ICECandidateTypeRelay
+			tunneled := false
+			properties := []string{}
+			candidateURLs := []string{}
+
+			for _, server := range servers {
+				if server.Username == wsnet.TURNProxyICECandidate().Username {
+					candidateURLs = append(candidateURLs, fmt.Sprintf("turns:%s", url.Host))
+					if !relay {
+						continue
+					}
+					tunneled = true
+					continue
+				}
+
+				candidateURLs = append(candidateURLs, server.URLs...)
+			}
+			properties = append(properties, fmt.Sprintf("candidates=%s", strings.Join(candidateURLs, ",")))
+
+			connectionText := "direct via STUN"
+			if relay {
+				connectionText = "proxied via TURN"
+			}
+			if tunneled {
+				connectionText = fmt.Sprintf("proxied via %s", url.Host)
+			}
+
+			fmt.Printf("%s %s %s (%s) %s\n",
+				color.New(color.Bold, color.FgWhite).Sprint("PING"),
+				workspace.Name,
+				color.New(color.Bold, color.FgGreen).Sprintf("connected in %.2fms", connectionMS),
+				connectionText,
+				strings.Join(properties, " "),
+			)
+
+			ticker := time.NewTicker(time.Second)
+			seq := 1
+			for {
```

Comment:
what do you think about adding a `--count` or `-c` option like `ping`, so that someone can quit automatically after a few pings? that way, the command can be used non-interactively too

Reply:
I like it... will add!

Reply:
Done!